### ISSUE TITLE
Uncaught (in promise) TypeError: given schema is no Schema-object

### ISSUE THREAD

**Issue by CrashyBang** [pubkey/rxdb#77]
## Case

Bug.

## Issue

I am getting the following error when using RxDB with electron, HOWEVER this is only happen when I build the app not when I run it in dev mode.

![selection_002](https://cloud.githubusercontent.com/assets/2978850/23828660/e8612914-073d-11e7-8309-0f2cb56ac03c.png)

## Info
- Environment: Electron
- Adapter: (ndexedDB

## Code

```js
console.group('Initialize database.');
console.info('Creating user database.');
const database = await RxDB.create({
	name: 'moderator',
	adapter: 'idb'
});
console.info('Created user database.');

database.waitForLeadership().then(() => {
	console.info('leader.');
});

console.info('Creating collections.');
await Promise.all(collections.map(colData => database.collection(colData)));
console.info('Created collections.');

console.log('Start sync.');
collections.filter(col => col.sync).map(col => col.name).map(colName => database[colName].sync(url + colName + '/'));

console.groupEnd();

return database;
```

**Comment by pubkey** [comment]
How do you build the app? There is currently no build-script

**Comment by CrashyBang** [comment]
Hey @pubkey,

I am using [electron-builder](https://github.com/electron-userland/electron-builder) relevant `package.json` info below:

```json
{
  "main": "electron/index.js",
  "scripts": {
    "app:prod": "build -wl",
  },
  "license": "MIT",
  "build": {
    "files": [
      "electron",
      "node_modules"
    ],
    "linux": {
      "target": [
        "AppImage"
      ]
    },
    "win": {
      "target": [
        "nsis"
      ]
    }
  },
  "devDependencies": {
    "electron": "^1.4.15",
    "electron-builder": "^14.1.1",
    "electron-devtools-installer": "^2.1.0",
  }
}
```

I am wondering if it may be something to do with webpack building the scripts but I am just using babel as per the examples so I can't see why that would effect it?

**Comment by CrashyBang** [comment]
Had a quick look at the source here is what babel is rendering:

```js
{
    name: "assignments",
    schema: {
        title: "Assignment Schema",
        description: "Describes an assignment",
        version: 0,
        type: "object",
        properties: {
            name: {
                type: "string",
                primary: !0
            },
            curriculum: {
                type: "string"
            },
            year: {
                type: "integer"
            }
        },
        required: ["curriculum", "year"]
    },
    sync: !0
}
```

**Comment by pubkey** [comment]
The schema from babel looks ok. I don't know how I should reproduce this error. Maybe you can edit our  electron-example in a way to reproduce the error.

**Comment by CrashyBang** [comment]
Hey @pubkey,

I had similar thoughts I will put together a _minimal working example_ hopefully tonight and that should let me know if it is some 3rd party lib interfering or if it actually an rxdb bug.

Will add a link to the repo once I have it up.

Cheers.

**Comment by pubkey** [comment]
Hi @CrashyBang  Any update on this?

**Comment by CrashyBang** [comment]
Hey @pubkey sorry have been slammed at work and haven't even had a chance to begin to debug this :L hope to be onto it some time this week.

Cheers.

**Comment by hippycore** [comment]
Same error with react-native, on release mode only

**Comment by matsuo-ts** [comment]
Same issue with @CrashyBang in my electron project. 
Writing the code with React and compile and bundle it with webpack.
This issue occurs when I minify the entire code for production.
No error if I bundling without minifying.

Here is the console error message.

```
Failed to open Database:  TypeError: given schema is no Schema-object
    at Object.$.a (RxCollection.js:609)
    at he.collection (RxDatabase.js:228)
```

**Comment by pubkey** [comment]
@matsuo-ts Can you debug into it and check what `constructor.name` is given [here](https://github.com/pubkey/rxdb/blob/d922f08f64015ef516f93f7c25a621d2d8676a6e/dist/lib/RxCollection.js#L46) ?

**Comment by pubkey** [comment]
Can't reproduce atm. We need a working build on one of [examples](https://github.com/pubkey/rxdb/tree/master/examples)

**Comment by pubkey** [comment]
A workarround is to enable `keep_fnames` in your uglifyJS-config.
Webpack-example:
```js
new webpack.optimize.UglifyJsPlugin({
                beautify: false,
                mangle: {
                    screw_ie8: true,
                    keep_fnames: true // <-- set this to true for quickfix
                }, //prod
                compress: { //prod
                    screw_ie8: true,
                    warnings: false
                },
                comments: false //prod
            })
```

**Comment by pubkey** [comment]
This is now fixed. Thx @muneebs 
The changed files will be published to npm with version 4.0.0.

If you need a fixed module faster, use [this](https://github.com/pubkey/rxdb/commit/ac079880872b8c8b90db6602c20c5c740374706f) commit-hash into your package.json

**Comment by matsuo-ts** [comment]
@pubkey Sorry for late reply and thank you for your advise. 

This
> A workarround is to enable keep_fnames in your uglifyJS-config.
Webpack-example:
>
> new webpack.optimize.UglifyJsPlugin({
                beautify: false,
                mangle: {
                    screw_ie8: true,
                    keep_fnames: true // <-- set this to true for quickfix
                }, //prod
                compress: { //prod
                    screw_ie8: true,
                    warnings: false
                },
                comments: false //prod
            })

works fine on version `3.0.8` but after update to `4.0.0` I encountered another error.
Let me report that on the new issue.
